### PR TITLE
Allow configuring captcha by custom flow

### PR DESCRIPTION
**Pull request by pkong-ds**
ref dev-1606

Allow below configuration

```
bot_protection:
  requirements:
    signup_or_login:
      mode: never
    account_recovery:
      mode: always
    password:
      mode: never
    oob_otp_email:
      mode: always
    oob_otp_sms:
      mode: always
```

### CODE REVIEW

**Comment by pkong-ds** [comment]
@tung2744 Can help review (and merge) this PR? 🙇 Seems daic too many PRs already :sadfrog:

**Comment by louischan-oursky** [comment]
Let me review this. @tung2744 please review https://github.com/authgear/authgear-server/pull/4505 for me

**Comment by pkong-ds** [comment]
Also added new generated flows

**Comment by louischan-oursky** [comment]
The e2e test failed.

**Review by louischan-oursky — CHANGES_REQUESTED** [review]
The e2e tests failed.

**Comment by pkong-ds** [comment]
@louischan-oursky e2e tests pass locally now, thanks!

File: pkg/lib/config/testdata/bot_protection_tests.yaml [hunk 1]
```diff
@@ -64,3 +64,67 @@ name: enabled-false
...
+  /requirements/signup_or_login/mode: enum
+    map[actual:invalid_mode expected:[never always]]
+value:
+  enabled: true
+  provider:
+    type: cloudflare
+    site_key: dummy
+  requirements:
+    signup_or_login:
+      mode: invalid_mode
```
Comment (louischan-oursky) [inline pkg/lib/config/testdata/bot_protection_tests.yaml:130]: Please keep the newline at the end of the file. Configure your text editor to do this for text files.

**Review by louischan-oursky — APPROVED** [review]

**Pull request merged**